src/api-serverless/src/proxies/proxy.api.service.ts

```diff
@@ -199,7 +207,48 @@ export class ProfileProxyApiService {
       count,
       page: page,
       next: profileProxies.length === page_size,
-      data: profileProxies
+      data: await this.profileProxiesMapper.profileProxyEntitiesToApiProfileProxies(
+        {
+          profileProxyEntities: profileProxies
+        }
+      )
+    };
+  }
+
+  async getProfileGrantedProfileProxies({
+    created_by,
+    page,
+    page_size,
+    sort,
+    sort_direction
+  }: {
+    readonly created_by: string;
+    readonly page: number;
+    readonly page_size: number;
+    readonly sort: string;
+    readonly sort_direction: string;
+  }): Promise<Page<ProfileProxy>> {
+    const [profileProxies, count] = await Promise.all([
+      this.profileProxiesDb.findProfileGrantedProfileProxies({
+        created_by,
+        page,
+        page_size,
+        sort,
+        sort_direction
+      }),
+      this.profileProxiesDb.countProfileGrantedProfileProxies({
+        created_by
+      })
+    ]);
+    return {
+      count,
+      page: page,
+      next: profileProxies.length === page_size,
```

Comment:
this should be done by count. if you do it like this, you could send a false flag, if last page is exactly page size